config-service-api/src/main/proto/org/hypertrace/config/service/v1/config_service.proto

```diff
@@ -161,3 +164,39 @@ message UpsertAllConfigsResponse {
     optional google.protobuf.Value prev_config = 5;
   }
 }
+
+message Filter {
+  oneof type {
+    RelationalFilter relational_filter = 1;
+    LogicalFilter logical_filter = 2;
+  }
+}
+
+message RelationalFilter {
+  string lhs = 1;
+  RelationalOperator op = 2;
+  google.protobuf.Value rhs = 3;
+}
+
+message LogicalFilter {
+  LogicalOperator op = 1;
+  repeated Filter operands = 2;
+}
+
+enum RelationalOperator {
+  RELATIONAL_OPERATOR_UNSPECIFIED = 0;
+  RELATIONAL_OPERATOR_EQ = 1;
+  RELATIONAL_OPERATOR_NEQ = 2;
+  RELATIONAL_OPERATOR_IN = 3;
+  RELATIONAL_OPERATOR_NOT_IN = 4;
+  RELATIONAL_OPERATOR_GT = 5;
+  RELATIONAL_OPERATOR_LT = 6;
+  RELATIONAL_OPERATOR_GE = 7;
```

Comment:
nit - GTE is a bit more readable, assuming this is greater than or equal to. Not trying to save chars here.